steps/24_categorical_support/step.md

```diff
@@ -0,0 +1,143 @@
+# Adding categorical support to sktime
+
+Contributors: @Abhay-Lejith
+
+## Introduction
+
+### Objectives 
+Allow users to pass data with categorical features to any estimator. If not supported, return an informative error message. Otherwise it should run without errors.
+
+### Current status:
+* Non-numeric dtypes are completely blocked off in datatype checks.
+* https://github.com/sktime/sktime/pull/5886 - This pr removes these checks but has not been merged yet as most estimators do not support categorical and will break without informative error messages.
+* https://github.com/sktime/sktime/pull/6490 - pr adding feature_kind metadata
+
+
+## Description of proposed designs
+
+
+For each design we will have to discuss 8 combinations of cases:
+1. estimator natively supports categorical: yes/no
+2. categorical feature specified by user: yes/no
+3. categorical present in data: yes/no
+    
+## Specifying categorical features:
+- The reason I proposed this is in case the user wants to treat a column with numerical values as categorical. This will not be detected by feature_kind as it relies only on the dtype.
+- More on how this is tackled in each design below.
+---
+---
+
+
+### Design 1. Allow categorical features ONLY in those estimators which support natively. (no encoding internally)
+
+- `categorical_features` parameter will be present only in estimators that support natively.
+- For estimators that do not support natively, users can make a pipeline with an encoder of their choice and use categorical variables. This is the general ml workflow, like how it's done in sklearn too.
+    
+    #### How tag is going to be used
+    - in datatype checks :
+    ```
+    if categorical support tag is false:
+        if categorical feature is present:
+            error - cat feature is not supported by this estimator
+    ```       
+
+- In this case `categorical feature` will not be a parameter in models that do not support. So (NO,YES,_) is not possible.    
+
+| **Estimator supports categorical natively** | **Cat feature specified by user** | **Cat feature present in data** | **Outcome** | 
+|:-----------------:|:-------------------------:|:--------------:|:---------------:|
+| NO | NO | NO | normal case, proceeds as usual without error. |
+| NO | NO | YES| informative error message to user that categorical feature is not supported by the estimator |
+| NO | YES| NO | X |
+| NO | YES| YES| X |
+| YES| NO | NO | normal case, proceeds as usual without error. |
+| YES| NO | YES| use categorical columns as detected by feature_kind and proceed. |
+| YES| YES| NO | informative error message that specified column was not present in data.|
+| YES| YES| YES| use categorical columns as detected by feature_kind along with the user specified columns(union) and proceed. |
+
+- in case where specified column was not found, we can also choose to ignore and continue with warning that column was not found.
```

Comment:
Clarification: what do you mean by "continue" here?

For example, let's consider ARIMA for revenue forecasting and we have categorical feature of user rating. What will happen?

Reply:
@yarnabrina This is design 1 which allows categorical only in estimators which support natively. ARIMA does not support natively, so it would just fail saying categorical not supported.

Reply:
so, if we have a different approach, I would suggest you record it and try to structure the different choices one can make here

Reply:
@fkiraly , the document has three major headings that say Design 1, Design 2 and Design 3. Is it really not clear?